From 6e6747161b9d8b8827e9cfef516ffd13304fd805 Mon Sep 17 00:00:00 2001
From: Michal Hrusecky <michal.hrusecky@nic.cz>
Date: Tue, 9 Apr 2019 14:10:20 +0200
Subject: [PATCH] php8: Integrate php-fpm into lighttpd

We have lighttpd installed by default so making sure that installation of php
enables php sites.
---
 lang/php8/Makefile               | 3 +++
 lang/php8/files/php-fpm.lighttpd | 7 +++++++
 2 files changed, 10 insertions(+)
 create mode 100644 lang/php8/files/php-fpm.lighttpd

diff --git a/lang/php8/Makefile b/lang/php8/Makefile
index 03e0b21a0d..3960c4e927 100644
--- a/lang/php8/Makefile
+++ b/lang/php8/Makefile
@@ -551,6 +551,9 @@ define Package/php8-fpm/install
 
 	$(INSTALL_DIR) $(1)/etc/init.d
 	$(INSTALL_BIN) ./files/php8-fpm.init $(1)/etc/init.d/php8-fpm
+
+	$(INSTALL_DIR) $(1)/etc/lighttpd/conf.d/
+	$(INSTALL_DATA) ./files/php-fpm.lighttpd $(1)/etc/lighttpd/conf.d/30-php-fpm.conf
 endef
 
 define Package/php8-fpm/conffiles
diff --git a/lang/php8/files/php-fpm.lighttpd b/lang/php8/files/php-fpm.lighttpd
new file mode 100644
index 0000000000..7120346a13
--- /dev/null
+++ b/lang/php8/files/php-fpm.lighttpd
@@ -0,0 +1,7 @@
+fastcgi.server += ( ".php" =>
+        ((
+                "socket" => "/var/run/php8-fpm.sock",
+                "broken-scriptfilename" => "enable",
+                "fix-root-scriptname" => "enable"
+        ))
+)
-- 
2.38.1

